### PR TITLE
Switch model in query window

### PR DESCRIPTION
**Pull request by phlpsong**
close https://github.com/tisfeng/Easydict/issues/456

### CODE REVIEW

File: Easydict/Feature/ViewController/View/ResultView/EZResultView.m [hunk 1]
```diff
@@ -409,6 +415,30 @@ - (void)updateServiceModelLabel {
...
+        [menu addItem:item];
+    }
+    [menu popUpMenuPositioningItem:nil atLocation:[NSEvent mouseLocation] inView:nil];
+}
+
+- (void)modelDidSelected:(NSMenuItem *)sender {
+    EZOpenAILikeService *service = (EZOpenAILikeService *)self.result.service;
+    service.model = sender.title;
+    self.serviceModelButton.title = service.model;
+    [self postServiceUpdatedNotification:service.serviceType];
```
Comment (tisfeng) [inline Easydict/Feature/ViewController/View/ResultView/EZResultView.m:433]:
```suggestion
     if (![service.model isEqualToString:sender.title]) {
        service.model = sender.title;
        self.serviceModelButton.title = service.model;
        [self postServiceUpdatedNotification:service.serviceType];
    }
```

Comment (tisfeng) [reply Easydict/Feature/ViewController/View/ResultView/EZResultView.m:433]: If the selected model doesn't change, we don' have to update.

File: Easydict/Feature/Service/OpenAI/EZOpenAIService.m [hunk 2]
```diff
@@ -62,6 +62,14 @@ - (NSString *)model {
     return model;
 }
 
+- (void)setModel:(NSString *)model {
+    [[NSUserDefaults standardUserDefaults] setValue:model forKey:EZOpenAIModelKey];
+}
+
+- (NSArray<NSString *> *)availableModels {
+    return @[@"gpt-3.5-turbo-0125", @"gpt-4-0125-preview"];
```
Comment (tisfeng) [inline Easydict/Feature/Service/OpenAI/EZOpenAIService.m:70]: Can we optimize this code? It duplicates the code in the configuration, and these models will follow OpenAI changes later.

```swift
enum OpenAIModels: String, CaseIterable {
    case gpt3_5_turbo_0125 = "gpt-3.5-turbo-0125"
    case gpt4_0125_preview = "gpt-4-0125-preview"
}
```

Comment (phlpsong) [reply Easydict/Feature/Service/OpenAI/EZOpenAIService.m]: You're right, I have rewritten the `OpenAIService` to reuse this enum type.

Comment (tisfeng) [reply Easydict/Feature/Service/OpenAI/EZOpenAIService.m]: ok

**Review by tisfeng — APPROVED** [review]
Looks good.

**Comment by tisfeng** [comment]
Please review this PR.

**Review by Jerry23011 — COMMENTED** [review]
This popup looks like one with a mouse right click, is it possible that we can have a fixed position for the popup?
截屏2024-03-21 14 00 23

**Comment by phlpsong** [comment]
Could you share some examples here? I'm not quite sure what you meant by the fixed position.

**Comment by Jerry23011** [comment]
Certainly, the popups in settings have a fixed position no matter where I click on them. Please see recording below

https://github.com/tisfeng/Easydict/assets/89069957/06b75f98-cf8f-4839-9e5b-321616abcd9c

**Comment by tisfeng** [comment]
Yes, the NSMenu should be displayed in a fixed position, e.g. clicking on the bottom left corner of the view, which looks best.

image

**Comment by tisfeng** [comment]
I didn't look at the detailed reason why the atLocation coordinates would be drifting, it seems a little strange.

Also, I noticed that NSMenu is used in other places, such as `EZSelectLanguageButton`,  we can write a Category for NSMenu, making it convenient to use to show NSMenu at the bottom left of a view.

**Comment by phlpsong** [comment]
Thanks, I got you. 👍

But personally, the Picker-like component is not suitable in this case, popup the menu from the left corner looks better.

**Comment by tisfeng** [comment]
ok, we don't need to change the system Picker, just the NSMenu.

**Review by tisfeng — APPROVED** [review]

**Review by Jerry23011 — APPROVED** [review]
LGTM

**Pull request merged**